apps/wellknown/views/application.py

```diff
@@ -0,0 +1,29 @@
+from ...dot_ext.models import Application
+from rest_framework.generics import ListAPIView
+from ..serializers import ApplicationListSerializer
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.permissions import AllowAny
+from rest_framework.renderers import JSONRenderer
+
+
+class ApplicationListPagination(PageNumberPagination):
+    page_size = 10
+    page_size_query_param = 'page_size'
+    max_page_size = 10000
+
+
+class ApplicationListView(ListAPIView):
+    """
+    View to provide an applications list.
+
+    An application that has active=False or with a label name = "internal-use"
+    will be excluded from this view.
+    """
+    permission_classes = (AllowAny,)
+    renderer_classes = (JSONRenderer,)
+    serializer_class = ApplicationListSerializer
+    pagination_class = ApplicationListPagination
+
+    def get_queryset(self):
+        queryset = Application.objects.exclude(active=False).exclude(applicationlabel__name='internal-use').order_by('name')
```

Comment:
Define the label to exclude externally (a django setting?).
Also, if possible, make it a blacklist array such that if `applicationlabel__slug__in=blacklist` the application is excluded.

Reply:
This is a great idea!
I added a django setting to be passed in externally during deployment with the following default values:
DJANGO_APP_LIST_EXCLUDE="['internal-use', 'do-not-publish']"